torch/testing/_internal/common_utils.py

```diff
@@ -983,24 +992,29 @@ def wrapper(*args, **kwargs):
             fn(*args, **kwargs)
     return wrapper
 
-# Skips a test on CUDA if ROCm is unavailable or its version is lower than requested.
-def skipIfRocmVersionLessThan(version=None):
+# Skips a test on CUDA if ROCm is available and its version is lower than requested.
+def skipIfRocmAndVersionLessThan(version=None):
     def dec_fn(fn):
         @wraps(fn)
         def wrap_fn(self, *args, **kwargs):
-            if not TEST_WITH_ROCM:
-                reason = "ROCm not available"
-                raise unittest.SkipTest(reason)
-            rocm_version = str(torch.version.hip)
-            rocm_version = rocm_version.split("-")[0]    # ignore git sha
-            rocm_version_tuple = tuple(int(x) for x in rocm_version.split("."))
-            if rocm_version_tuple is None or version is None or rocm_version_tuple < tuple(version):
-                reason = "ROCm {0} is available but {1} required".format(rocm_version_tuple, version)
-                raise unittest.SkipTest(reason)
+            if TEST_WITH_ROCM:
+                rocm_version = str(torch.version.hip)
+                rocm_version = rocm_version.split("-")[0]    # ignore git sha
+                rocm_version_tuple = tuple(int(x) for x in rocm_version.split("."))
+                if rocm_version_tuple is None or version is None or rocm_version_tuple < tuple(version):
+                    reason = "ROCm {0} is available but {1} required".format(rocm_version_tuple, version)
+                    raise unittest.SkipTest(reason)
             return fn(self, *args, **kwargs)
         return wrap_fn
     return dec_fn
 
+# Skips a test on CUDA if ROCm is unavailable or its version is lower than requested.
+def skipIfRocmVersionLessThan(version=None):
```

Comment:
I wonder if this should be fully removed? The only other usage is https://github.com/pytorch/pytorch/blob/03abcf231726b094bd378ead59d81c673fe11b7d/test/test_nn.py#L11396

Does anyone know if that test should or shouldn't run on "real" CUDA?

Reply:
I think it is ok to keep the original name and change the behavior.

Reply:
But the behavior is very unexpected, to say the least..
I hope `@skipIfRocm` does not skip tests on CUDA